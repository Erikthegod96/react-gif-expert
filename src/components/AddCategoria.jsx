import { useState } from "react"
import PropTypes from 'prop-types';

export const AddCategoria = ({ onNuevaCategoria }) => {

    const [inputValor, setInputValor] = useState('')

    const onInputChange = ({ target }) => {
        setInputValor(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault()
        if (inputValor.trim().length <= 1) return
        onNuevaCategoria(inputValor.trim())
        setInputValor('')
    }

    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input
                type='text'
                placeholder="Buscar gifs"
                value={inputValor}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategoria.propTypes = {
    onNuevaCategoria: PropTypes.func.isRequired
}
