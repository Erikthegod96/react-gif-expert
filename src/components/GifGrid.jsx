import { GifItem } from "./"
import { useFetchGifs } from "../hooks/useFetchGifs"
import PropTypes from 'prop-types'

export const GifGrid = ({ categoria }) => {

    const { imagenes, isLoading } = useFetchGifs(categoria)


        return (
            <>
                <h3>{categoria}</h3>
                {
                    isLoading && (<h2>Cargando...</h2>)
                }
                
                <div className="card-grid">
                    {
                        imagenes.map((imagen) => (
                            <GifItem
                                key={imagen.id}
                                {...imagen}
                            />
                        ))
                    }
                </div>
            </>
        )
}


GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}
