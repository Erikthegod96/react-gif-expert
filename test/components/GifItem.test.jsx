import {render,screen} from '@testing-library/react'
import { GifItem } from '../../src/components/GifItem'

describe('Prueba en GifItem', ()=>{

    const title = 'Gifs'
    const url = 'http://localhost/Goku'
    test('Debe hacer match snapshot', () => {
        const {container} = render (<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
        render (<GifItem title={title} url={url}/>)
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })
})