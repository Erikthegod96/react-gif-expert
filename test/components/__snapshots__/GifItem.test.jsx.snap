// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifItem Debe hacer match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gifs"
      src="http://localhost/Goku"
    />
    <p>
      Gifs
    </p>
  </div>
</div>
`;
